### PR TITLE
Feat/connect redux

### PR DESCRIPTION
**Pull request by ex3emxl**

### CODE REVIEW

File: src/app.component.js [hunk 1]
```diff
@@ -1,10 +1,33 @@
...
+        this.props.dispatch(check());
+    }
+
+    render() {
+
+        const { user, info } = this.props;
+
+        return (
+            <div className="wrapper">
+                <Header/>
```
Comment (iliatcymbal) [inline src/app.component.js]: use spase before `/>`

File: src/app.component.js [hunk 2]
```diff
@@ -1,10 +1,33 @@
...
+                <Main>
+                    <Pages user={ user } info={ info }/>
```
Comment (iliatcymbal) [inline src/app.component.js]: use spase before `/>`

File: src/app.component.js [hunk 3]
```diff
@@ -1,10 +1,33 @@
...
+                </Main>
+            </div>
+        );
+    }
+}
+
+const mapState = state => ({ user: state.user, info: state.info, error: state.error });
```
Comment (iliatcymbal) [inline src/app.component.js]: you don't need `error` here

File: src/components/main/Main.js [hunk 4]
```diff
@@ -1,43 +1,15 @@
...
-import { checkUser } from '../../services';
-import Login from '../../pages/login';
-import Dashboard from '../../pages/dashboard';
 
-class Main extends Component {
-    state = {
-        user: null,
-        loading: true
-    }
+export class Main extends Component {
```
Comment (iliatcymbal) [inline src/components/main/Main.js]: it should be a pure component, not a class

File: src/pages/Pages.js [hunk 5]
```diff
@@ -1,11 +1,23 @@
...
+import Dashboard from './dashboard';
+import Login from '../pages/login';
 import Tasklist from '../components/tasklist';
 
-export const Pages = () => (
+export const Pages = ({user}) => (
     <Switch>
-        <Route path="/tasklist" component={ Tasklist } key="tasklist"/>
-        <Route path="/" exact component={ Main } key="home"/>
+        {console.log(user)}
```
Comment (iliatcymbal) [inline src/pages/Pages.js]: remove `console`

File: src/pages/Pages.js [hunk 6]
```diff
@@ -1,11 +1,23 @@
...
+        {
+            user ? [
+                <Route path="/tasklist" component={ Tasklist } key="tasklist"/>,
+                <Route path="/"
+                       render={() => <Dashboard user={user}/>}
+                       key="dashboard"/>
+            ] : [
+                <Route path="/" exact component={ Login } key="login"/>
```
Comment (iliatcymbal) [inline src/pages/Pages.js:17]: why here do you have `exact` attribute and do not set it for ather route with the same path (13 line)?

File: src/pages/dashboard/Dashboard.js [hunk 7]
```diff
@@ -1,27 +1,23 @@
 import React from 'react';
+import PropTypes from 'prop-types';
+
 import Hello from "../../components/hello";
```
Comment (iliatcymbal) [inline src/pages/dashboard/Dashboard.js]: use single quotes

File: src/pages/login/Login.js [hunk 8]
```diff
@@ -1,15 +1,15 @@
-import { login } from '../../services';
+import { connect } from 'react-redux';
+import { login } from '../../store/user';
 
-const Login = ({ onLogin }) => {
+const Login = ({ dispatch }) => {
     const onSubmit = (ev) => {
```
Comment (iliatcymbal) [inline src/pages/login/Login.js]: use `e` or `event` but not `ev`

File: src/store/user/actions.js [hunk 9]
```diff
@@ -0,0 +1,8 @@
+export const LOGIN = 'Login';
+export const login = data => ({type: LOGIN, data});
```
Comment (iliatcymbal) [inline src/store/user/actions.js]: use spaces after `{` and before `}`

File: src/store/user/actions.js [hunk 10]
```diff
@@ -0,0 +1,8 @@
...
+
+export const CHECK_USER = 'Check user';
+export const check = () => ({type: CHECK_USER});
```
Comment (iliatcymbal) [inline src/store/user/actions.js]: use spaces after `{` and before `}`

File: src/store/user/actions.js [hunk 11]
```diff
@@ -0,0 +1,8 @@
...
+
+export const SET_USER = 'Set user';
+export const setUser = data => ({type: SET_USER, data});
```
Comment (iliatcymbal) [inline src/store/user/actions.js]: use spaces after `{` and before `}`

File: src/store/user/sagas.js [hunk 12]
```diff
@@ -0,0 +1,35 @@
...
+import { checkUser, login } from "../../services";
+
+import {
+    CHECK_USER,
+    LOGIN,
+    setUser
+} from "./actions";
+
+function* check() {
+    let user;
```
Comment (iliatcymbal) [inline src/store/user/sagas.js]: make it as a `const` and move to `try` block

**Review by iliatcymbal — CHANGES_REQUESTED** [review]
Looks pretty good, but need some changes

File: src/store/user/sagas.js (hunk 12 above)
Comment (ex3emxl) [reply src/store/user/sagas.js]: @iliatcymbal This is generate [syntax error](https://prnt.sc/mo3zsr)

File: src/components/main/Main.js [hunk 13]
```diff
@@ -2,14 +2,10 @@ import React from 'react';
 
 import './main.scss';
 
-export class Main extends Component {
-
-  render() {
-    return (
+export const Main = (props) =>{
+  return (
```
Comment (iliatcymbal) [inline src/components/main/Main.js:6]: please, use short record for arrow function (without return)

File: src/pages/Pages.js [hunk 14]
```diff
@@ -6,11 +6,11 @@ import Tasklist from '../components/tasklist';
...
 export const Pages = ({user}) => (
     <Switch>
-        {console.log(user)}
         {
             user ? [
                 <Route path="/tasklist" component={ Tasklist } key="tasklist"/>,
                 <Route path="/"
+                       exact
                        render={() => <Dashboard user={user}/>}
                        key="dashboard"/>
```
Comment (iliatcymbal) [inline src/pages/Pages.js:15]: space before `/>`; it should be everywhere

**Review by iliatcymbal — APPROVED** [review]

**Pull request merged**